.github/workflows/build-webapi.yaml

```diff
@@ -9,27 +9,45 @@ on:
     branches: [ dev, main ]
     paths:
       - 'src/dotnet/**'
+      - '.github/workflows/**'
 
 jobs:
-  build:
+  container-build-step:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: ./src/dotnet/CarbonAware.WebApi
     steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v2
+    
+    - name: Docker Target Build
+      run: docker build --target=build . -f CarbonAware.WebApi/Dockerfile -t ca-api-build
+      working-directory: ./src/dotnet
+
+    - name: Docker Unit Test
+      run: docker run -w //src/ ca-api-build dotnet test
+      working-directory: ./src/dotnet
 
-    - uses: actions/checkout@v2
-    - name: Setup .NET
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: 6.0.x
+  container-publish-step:
+      runs-on: ubuntu-latest
+      steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
 
-    - name: Restore dependencies
-      run: dotnet restore
+      - name: Docker Target Final
+        run: docker build . -f CarbonAware.WebApi/Dockerfile -t ca-api
+        working-directory: ./src/dotnet
+        
+      - name: Docker Run Container
+        run: |
+          docker run -d --name publish-container -p 8080:80 ca-api
+          docker container ls
+          
+      - name: Docker Wget in container
```

Comment:
well, I think you're correct @bderusha that it isn't a unit-test, but rather a 'smoke test' to be sure the web-app actually runs-without-error 'a least once' after being built in the container, since we'd seen in the older version of DockerFile (from upstream) wasn't the case.

traditionally we thought we would also want to ensure the build container 'lgtm' before it was published as a build-artifact (or published to a registry)

I think what is in place is still "the right 1st step", and no-one was expecting the pipelines to not have to iterate in the future -- shall we discuss more offline?